### ISSUE TITLE
Deep stack traces are truncated in the Debug Console in VS Code

### ISSUE THREAD

**Issue by DonJayamanne** [Microsoft/ptvsd#975]
_From @gvellut on October 25, 2018 15:53_

## Environment data

- VS Code version: 1.28.1
- Extension version: 2018.9.1 (18 Oct 2018)
- OS and version: Windows 10
- Python version: Standard Python 3.7
- Type of virtual environment used (N/A | venv | virtualenv | conda | ...): venv 

## Actual behavior

I launch a python program using the following configuration:

>  {
>             "name": "Python: App",
>             "type": "python",
>             "request": "launch",
>             "module": "app",
>             "console": "none",
>             "env": {
>                 "FLASK_APP": "app",
>                 "FLASK_ENV": "development",
>                 "FLASK_DEBUG": false,
>                 "DATABASE_URI": "mysql+pymysql://root:admin@localhost:3307/platform",
>             }
>         }

The output of deep stack traces is truncated in the Debug Console. For example:

> 2018-10-25 17:35:10,313 flask.app    ERROR    Exception on /test_rio [GET]
> Traceback (most recent call last):
>   File "C:\Users\gvellut\Documents\Anak\bt2\.venv\lib\site-packages\flask\app.py", line 2292, in wsgi_app
>     response = self.full_dispatch_request()
>   File "C:\Users\gvellut\Documents\Anak\bt2\.venv\lib\site-packages\flask\app.py", line 1815, in full_dispatch_request
>     rv = self.handle_user_exception(e)
>   File "C:\Users\gvellut\Documents\Anak\bt2\.venv\lib\site-packages\flask\app.py", line 1718, in handle_user_exception
>     reraise(exc_type, exc_value, tb)
>   File "C:\Users\gvellut\Documents\Anak\bt2\.venv\lib\site-packages\flask\_compat.py", line 35, in reraise
>     raise value
>   File "C:\Users\gvellut\Documents\Anak\bt2\.venv\lib\site-packages\flask\app.py", line 1813, in full_dispatch_request
>     rv = self.dispatch_request()
>   File "C:\Users\gvellut\Documents\Anak\bt2\.venv\lib\site-packages\flask\app.py", line 1799, in dispatch_request
>     return self.view_func...

I saw this problem mentioned in the VSCode project https://github.com/Microsoft/vscode/issues/45860 but with no resolution. The issue mentions it comes from the vscode-python extension.

## Expected  behavior

The full output should be displayed (or a configuration option should be provided to enable it). For example, here is the output when launched outside VSCode:

> 2018-10-25 17:38:35,648 flask.app    ERROR    Exception on /test_rio [GET]
> Traceback (most recent call last):
>   File "C:\Users\gvellut\Documents\Anak\bt2\.venv\lib\site-packages\flask\app.py", line 2292, in wsgi_app
>     response = self.full_dispatch_request()
>   File "C:\Users\gvellut\Documents\Anak\bt2\.venv\lib\site-packages\flask\app.py", line 1815, in full_dispatch_request
>     rv = self.handle_user_exception(e)
>   File "C:\Users\gvellut\Documents\Anak\bt2\.venv\lib\site-packages\flask\app.py", line 1718, in handle_user_exception
>     reraise(exc_type, exc_value, tb)
>   File "C:\Users\gvellut\Documents\Anak\bt2\.venv\lib\site-packages\flask\_compat.py", line 35, in reraise
>     raise value
>   File "C:\Users\gvellut\Documents\Anak\bt2\.venv\lib\site-packages\flask\app.py", line 1813, in full_dispatch_request
>     rv = self.dispatch_request()
>   File "C:\Users\gvellut\Documents\Anak\bt2\.venv\lib\site-packages\flask\app.py", line 1799, in dispatch_request
>     return self.view_functions[rule.endpoint](**req.view_args)
>   File "C:\Users\gvellut\Documents\Anak\bt2\app.py", line 58, in test_rasterio
>     with rasterio.open("/vsigs/{}".format(url)) as src:
>   File "C:\Users\gvellut\Documents\Anak\bt2\.venv\lib\site-packages\rasterio\env.py", line 402, in wrapper
>     return f(*args, **kwds)
>   File "C:\Users\gvellut\Documents\Anak\bt2\.venv\lib\site-packages\rasterio\__init__.py", line 225, in open
>     s = DatasetReader(path, driver=driver, **kwargs)
>   File "rasterio/_base.pyx", line 199, in rasterio._base.DatasetBase.__init__
>   File "rasterio/_shim.pyx", line 64, in rasterio._shim.open_dataset
>   File "rasterio/_err.pyx", line 188, in rasterio._err.exc_wrap_pointer
> rasterio._err.CPLE_AppDefinedError: Fetching OAuth2 access code from auth code failed.
> 

## Steps to reproduce

The most basic Flask app that throws an exception will show the issue:

```
app = Flask(__name__)

@app.route("/test_trunc")
def test_trunc():
    raise Error("azeaez")

app.run()
```

_Copied from original issue: Microsoft/vscode-python#3015_

**Comment by DonJayamanne** [comment]
@karthiknadig 
I'm able to replicate this, here's the output from the JSON:
```js
{"type": "event", "seq": 18, "event": "output", "body": {"category": "stderr", "output": "Error on request:\nTraceback (most recent call last):\n  File \"/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/werkzeug/serving.py\", line 270, in run_wsgi\n    execute(self.server.app)\n  File \"/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/werkzeug/serving.py\", line 258, in execute\n    application_iter = app(environ, start_response)\n  File \"/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/flask/app.py\", line 1997, in __call__\n    return self.wsgi_app(environ, start_response)\n  File \"/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/flask/app.py\", line 1985, in wsgi_app\n    response = self.handle_exception(e)\n  File \"/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/flask/app.py\", line 1540, in handle_exception\n    reraise(exc_type, exc_value, tb)\n  File \"/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/flask/_compat...."}}
To Client:
Content-Length: 95
```

**Comment by karthiknadig** [comment]
This is due to `MAX_IO_MSG_SIZE` in `pydevd`. Do we want to make this a setting? If yes, then should it be called as and how will it be passed? If no, then what should be the default. 
/cc: @fabioz , @qubitron

**Comment by qubitron** [comment]
It seems like an obscure setting to add/maintain. I'm not sure what the ramifications are: does it need to be limited at all? Could we set the default high enough that users won't practically hit the limit?

**Comment by karthiknadig** [comment]
Looking at the code, and comments. The limit exists for performance reasons, @fabioz may know the impact of increasing this.

**Comment by fabioz** [comment]
@karthiknadig it should be safe to raise this limit at this point (it has been that same value for a *really* long time, so, the experiments I did back when that number was defined are no longer applicable). 

I think that the main thing for the limit right now is the case where the user prints something too big (i.e.: dumping a 50 MB png/array to stdout, so, making the needed data transforms and sending the data over the wire when remote debugging could have really bad implications where the debugger could be stuck for quite a while).

My suggestion is setting something as a 100kb limit (although this needs some manual testing dumping some really big strings to stdout/stderr to see if the debugger/client remain responsive).

**Comment by karthiknadig** [comment]
@fabioz I tried increasing this to 100,000. It looks like there is a cutoff at 10,764. Not sure if this is a default for the buffers. @int19h any idea about this?

**Comment by karthiknadig** [comment]
Closed via #1139. Sets the max io to 10k for now.

**Comment by ajay9022** [comment]
> @karthiknadig
> I'm able to replicate this, here's the output from the JSON:
> 
> ```js
> {"type": "event", "seq": 18, "event": "output", "body": {"category": "stderr", "output": "Error on request:\nTraceback (most recent call last):\n  File \"/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/werkzeug/serving.py\", line 270, in run_wsgi\n    execute(self.server.app)\n  File \"/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/werkzeug/serving.py\", line 258, in execute\n    application_iter = app(environ, start_response)\n  File \"/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/flask/app.py\", line 1997, in __call__\n    return self.wsgi_app(environ, start_response)\n  File \"/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/flask/app.py\", line 1985, in wsgi_app\n    response = self.handle_exception(e)\n  File \"/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/flask/app.py\", line 1540, in handle_exception\n    reraise(exc_type, exc_value, tb)\n  File \"/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/flask/_compat...."}}
> To Client:
> Content-Length: 95
> ```

Does this resolve for `OUTPUT` window too? If YES, then where to find the json file you are talking about?
